client-docs/sync.md

```diff
@@ -0,0 +1,179 @@
+# Sync API
+
+## Table of Contents
+
+- [Description](#description)
+
+- [Types](#types)
+
+  - [`ConnectionType`](#connectiontype)
+  - [`SyncInfo`](#syncinfo)
+
+- [Methods](#methods)
+
+  - [`start`](#start)
+  - [`stop`](#stop)
+  - [`info`](#info)
+  - [`enableDiscovery`](#enablediscovery)
+  - [`disableDiscovery`](#disablediscovery)
+  - [`enableSync`](#enablesync)
+  - [`disableSync`](#disablesync)
+
+## Description
+
+Provides an imperative API for managing project discovery and sync strategies.
+
+## Types
+
+### `ConnectionType`
+
+Represents the connection type used for discovery or sync.
+
+```ts
+type ConnectionType = "lan" | "internet";
+```
+
+### `SyncInfo`
+
+Information about the kind of sync that is enabled.
+
+```ts
+type SyncInfo = {
+  discovery: ConnectionType[];
+  sync: ConnectionType[];
+};
+```
+
+## Methods
+
+### `start`
```

Comment:
Not sure if these start and stop methods are necessary with the `enableDiscovery` and `enableSync` methods?

As I understand it:

`enableDiscovery` = "start looking for and connecting to other devices" (but don't start syncing with them)
`enableSync` = "start syncing with connected devices"
`disableDiscovery` = "stop looking for and connecting to other devices" (but remain connected to devices already connected?)
`disableSync` = "stop syncing with connected devices" (but should this wait for sync to complete first?)

I guess I'm unsure about what should actually cause disconnection from other clients? Maybe that's why we might need `stop()`? I'm still unsure about this API in my own head.

Reply:
hm yeah this was something i was struggling to come to an answer with. not sure if any clearer, but another approach would be:

1. remove `start` and `stop` methods such that `enable_` and `disable_` implicitly handle start/stop. in the case of disconnecting from other clients, i think converting to `set_` like in your other comment would be more conducive for this because then it's clearer when application code wants to stop since there's no concept of merged values.

2. add listenable events for discovery and sync i.e. `'discovery:start`', `'discovery:stop'`, `'sync:start'`, `'sync:stop'`

again, not sure if this is any clearer but can imagine that it's helpful to know when sync/discovery processes start and stop from an application perspective.

Reply:
updated via dc979e4c19ddde7dfe4009ec6c516d0815d0e1d0 with the alternative approach to use `set_` and add some events. think i like it more but curious to get thoughts